lib/workload/stateless/stacks/icav2-event-translator/README.md

```diff
@@ -0,0 +1,156 @@
+    "timeCreated": "2024-00-25T00:04:00Z",
+    "timeModified": "2024-00-25T00:07:00Z",
+    "userReference": "xxxxxx_xxxxxx_xxxxxx_xxxxxx_xxx_xxxxxxxxxx",
+    ...
+    "pipeline": {
+      "id": "bxxxx-cxxx-4xxx-8xxxx-axxxxxxxxxxx",
+      "timeCreated": "2024-00-00T01:16:50Z",
+      "timeModified": "2024-00-00T02:08:46Z",
+      "code": "BclConvert v0_0_0",
+      "urn": "urn:ilmn:ica:pipeline:bxxxx-cxxx-4xxx-8xxxx-axxxxxxxxxxx#BclConvert_v0_0_0",
+      "description": "This is an autolaunch BclConvert pipeline for use by the metaworkflow",
+      ...
+    },
+    ....
+  }
+}
+```
+
+## Outputs
+
+The AWS lambda functions put translated event following  ```WorkflowRunStateChange``` schema with the following event detail::
+
+```json5
+{
+  "version": "0", // by default is 0
+  "id": "dxxxxx-6xxxx-fxxx-1xxx-5xxxxx",
+  "detail-type": "WorkflowRunStateChange", // event type
+  "source": "orcabus.bcm", // from bcl-convert-manager service
+  "account": "xxxxxxxxxxxx",
+  "time": "2024-00-00T02:08:46Z",
+  "region": "ap-southeast-x",
+  "resources": [],
+  "detail": {
+    "portalRunId": '20xxxxxxxxxx',
+    "timestamp": "2024-00-25T00:07:00Z",
+    "status": "SUCCEEDED",
+    "workflowType": "bssh_bcl_convert",
+    "workflowVersion": "4.2.7",
+    "payload": {
+      "refId": None,
+      "version": "0.1.0",
+      "projectId": "valid_project_id",
+      "analysisId": "valid_payload_id",
+      "userReference": "123456_A1234_0000_TestingPattern",
+      "timeCreated": "2024-01-01T00:11:35Z",
+      "timeModified": "2024-01-01T01:24:29Z",
+      "pipelineId": "valid_pipeline_id",
+      "pipelineCode": "BclConvert v0_0_0",
+      "pipelineDescription": "BclConvert pipeline.",
+      "pipelineUrn": "urn:ilmn:ica:pipeline:123456-abcd-efghi-1234-acdefg1234a#BclConvert_v0_0_0"
+    }
+  }
+}
+```
+
+## lambda function for event translation
+
+### Receive and Translate ICAv2 event to Orcabus internal event
+Combine the iput and anylsis result to generat the orcabus internal event.
+Event Pattern (filter rules) for event pipe:
+```json5
+{
+  account: [Stack.of(this).account],
+  detail: {
+    'ica-event': {
+    eventCode: ['ICA_EXEC_028'],
+    projectId: [{ exists: true }],
+    payload: {
+        pipeline: {
+          code: [{ prefix: 'BclConvert' }],
+        },
+      },
+    },
+  },
+}
+```
+
+### Save this translation process to dynamoDB
+Send the orignal input and orcabus event after translation to the dynamo db for audit or recording of the process. 
+
+Dynamodb format:
+| id | id_type | analysis_id | analysis_status | portal_run_id | db_uuid | original_external_event | translated_internal_ica_event | timestamp |
+| -------- | ---- | ------- | ------- | ---- | ------- | ------- | ------- |------- | 
+| 20xxxxxxxxxx | portal_run_id | dxxxxx-6xxxx-fxxx-1xxx-5xxxxx | | | 4xxxxx-4xxxx-4xxx-4xxx-4xxxxx | | | |
+| dxxxxx-6xxxx-fxxx-1xxx-5xxxxx | analysis_id |  |   |dxxxxx-6xxxx-fxxx-1xxx-5xxxxx | 4xxxxx-4xxxx-4xxx-4xxx-4xxxxx | | | |
```

Comment:
Is the `portal_run_id`  here should be `20xxxxxxxxxx` so it matched with the row above?

Reply:
fixed in the table